core/src/main/scala/codeartifact/CodeArtifact.scala

```diff
@@ -15,23 +12,6 @@ object CodeArtifact {
     passwd = token
   )
 
-  private def getAuthorizationTokenRequest(domain: String, owner: String) =
```

Comment:
These are not longer needed after removing support for adding the auth token in the `.sbt/credentials` file.

Reply:
Oh, so this implicitly supported standard AWS CLI creds and you're removing that support? This seems like you're taking something away that would help us when we eventually rely on this plugin more.

Reply:
I tend agree with Jeff on this - unless there is a great reason to remove that support we should probably leave it - and just make the other option more predominant in the docs.

Reply:
This is what the error looked like when it supported all of the different ways to populate the auth token:
```
software.amazon.awssdk.core.exception.SdkClientException: Unable to load region from any of the providers in the chain software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain@417393d6: [software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider@7b50ce7a: Unable to load region from system settings. Region must be specified either via environment variable (AWS_REGION) or  system property (aws.region)., software.amazon.awssdk.regions.providers.AwsProfileRegionProvider@60d74664: No region provided in profile: default, software.amazon.awssdk.regions.providers.InstanceProfileRegionProvider@160e0ada: Unable to contact EC2 metadata service.]
```

It is not clear what the next steps should be to fix this error. I could go to the readme to this repo but the average engineer might not even know that we have this plugin.

After removing this it now looks like this:

```
[warn] Unable to get AWS CodeArtifact auth token from the CODEARTIFACT_AUTH_TOKEN environment variable.
``` 

or

```
[error] java.lang.RuntimeException: Failed to publish to AWS Codeartifact because the auth token was not set in the CODEARTIFACT_AUTH_TOKEN environment variable.
[error] 	at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$2(CodeArtifactPlugin.scala:82)
[error] 	at scala.Option.getOrElse(Option.scala:189)
[error] 	at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$1(CodeArtifactPlugin.scala:81)
[error] 	at codeartifact.CodeArtifactPlugin$.$anonfun$publish0$1$adapted(CodeArtifactPlugin.scala:78)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] (codeArtifactPublish) Failed to publish to AWS Codeartifact because the auth token was not set in the CODEARTIFACT_AUTH_TOKEN environment variable.
```

Reply:
If we want to we could also show the user how to populate the CODEARTIFACT_AUTH_TOKEN env with when the warning/error is printed.

Reply:
I was going to suggest just tweaking the error message.

Reply:
I was able to add back using the `.sbt/credentials` and I have updated the warning/errors.

I tried adding back the `CodeArtifact.getAuthToken(codeArtifactRepo.value)` but it with the change that makes it so the error/warning only gets logged once. We should come back to this in the future.

Reply:
Sure - that sounds like a good future PR.